backend/danswer/db/models.py

```diff
@@ -384,10 +385,16 @@ class ConnectorCredentialPair(Base):
     # controls whether the documents indexed by this CC pair are visible to all
     # or if they are only visible to those with that are given explicit access
     # (e.g. via owning the credential or being a part of a group that is given access)
-    is_public: Mapped[bool] = mapped_column(
-        Boolean,
-        default=True,
-        nullable=False,
+    access_type: Mapped[AccessType] = mapped_column(
```

Comment:
- access_type is now enum instead of boolean
- auto_sync_options is a dict that can flexibly store information needed to run a sync job
- last_time_perm_sync is used to manage sync jobs